### ISSUE TITLE
Rule proposal: vue/padding-line-between-blocks

### ISSUE THREAD

**Issue by posva** [vuejs/eslint-plugin-vue#855]
**Please describe what the rule should do:**

Enforce empty lines between blocks (`template`, `script`, `style`, and any other root level block)

**What category should the rule belong to?**

- [X] Enforces code style
- [ ] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- BAD -->
<template>
  <div></div>
</template>
<script>
export default {}
</script>
<custom></custom>
```

```vue
<!-- GOOD -->
<template>
  <div></div>
</template>

<script>
export default {}
</script>

<custom></custom>
```

**Additional context**

Having empty lines between blocks improves readability and allows quick navigation per paragraph (`{`/`}` in Vim)

similar to https://eslint.org/docs/rules/padding-line-between-statements#rule-details

**Comment by nathanchase** [comment]
+1 for this rule! Additionally, I'd like to see it implemented within any given element in a Vue template, like so:
```
<!-- GOOD -->
<template>
    <div>
        <p>Item 1</p>

        <p>Item 2</p>
    </div>
</template
```
```
<!-- BAD -->
<template>
    <div>
        <p>Item 1</p>
        <p>Item 2</p>
    </div>
</template
```

**Comment by posva** [comment]
@nathanchase That should definitely go  in another rule as it there are many ways of configuring it. I even wonder if there isn't already something

**Comment by nathanchase** [comment]
> @nathanchase That should definitely go in another rule as it there are many ways of configuring it. I even wonder if there isn't already something

I've looked, and surprisingly can't find anything within ESLint that applies specifically to ensuring newlines between elements in a Vue template.

**Comment by EmilyRosina** [comment]
:heavy_plus_sign: for the rule addition AND to separate @nathanchase's suggestion to another rule, as they're not the same thing :smile_cat: but i also could not find any rules to satisfy either

**Comment by JacobParis** [comment]
This rule is similar to @nathanchase's suggestion and would make a good model for the new rule

https://eslint.org/docs/rules/lines-between-class-members

**Comment by gerasimvol** [comment]
same for script options, please 🙏

```
// BAD
export default {
  name: 'ComponentName',
  components: { .... }
}
```

```
// GOOD
export default {
  name: 'ComponentName',

  components: { .... }
}
```

**Comment by milewski** [comment]
Please also padd the root tag itself -->

```vue
<!-- GOOD -->
<template>

  <div>

      <div>Demo 1</div>

      <div>Demo 2</div>

  </div>

</template>
```

```vue
<!-- BAD -->
<template>
  <div>
      <div>Demo1</div>
      <div>Demo2</div>
  </div>
</template>
```

**Comment by jamesdixon** [comment]
Is there an analogous option for script methods? I'm badlyyyyyy in need of this 😄

**Comment by juanbrujo** [comment]
https://eslint.vuejs.org/rules/padding-line-between-blocks.html
